scheduler/scheduler.go

```diff
@@ -139,22 +146,87 @@ func spawnHosts(ctx context.Context, newHostsNeeded map[string]int) (map[string]
 			continue
 		}
 
+		if ctx.Err() != nil {
+			return nil, errors.New("scheduling run canceled.")
+		}
+
+		d, err := distro.FindOne(distro.ById(distroId))
+		if err != nil {
+			grip.Error(message.WrapError(err, message.Fields{
+				"distro":  distroId,
+				"runner":  RunnerName,
+				"message": "failed to find distro",
+			}))
+			continue
+		}
+
+		CurrentParents, err := host.FindAllRunningParents()
```

Comment:
I would move these 2 queries out of the loop and only run if the distro supports containers, so that we only run them when necessary